src/legacy/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/edit_index_pattern.js

```diff
@@ -156,28 +159,27 @@ function destroyIndexedFieldsTable() {
 uiRoutes
   .when('/management/kibana/indices/:indexPatternId', {
     template,
+    k7Breadcrumbs: getEditBreadcrumbs,
     resolve: {
       indexPattern: function ($route, redirectWhenMissing, indexPatterns) {
         return indexPatterns
           .get($route.current.params.indexPatternId)
           .catch(redirectWhenMissing('/management/kibana/index'));
       }
-    }
+    },
   });
 
 uiRoutes
   .when('/management/kibana/indices', {
-    resolve: {
-      redirect: function ($location, config) {
-        const defaultIndex = config.get('defaultIndex');
-        let path = '/management/kibana/index';
+    redirectTo() {
+      const defaultIndex = chrome.getUiSettingsClient().get('defaultIndex');
+      let path = '/management/kibana/index';
 
-        if (defaultIndex) {
-          path = `/management/kibana/indices/${defaultIndex}`;
-        }
-
-        $location.path(path).replace();
+      if (defaultIndex) {
+        path = `/management/kibana/indices/${defaultIndex}`;
       }
+
+      return path;
```

Comment:
nit: seems a bit simpler to return the string rather than setting to `path`

```js
if (defaultIndex) {
  return ...;
} else {
  return ...;
}
```